framework/test/unit/modules/validators/endpoint.spec.ts

```diff
@@ -84,6 +84,55 @@ describe('ValidatorsModuleEndpoint', () => {
 					valid: true,
 				});
 			});
+
+			it('should resolve with false when proof of possession is invalid but bls key has a valid length', async () => {
+				const validPk =
+					'a491d1b0ecd9bb917989f0e74f0dea0422eac4a873e5e2644f368dffb9a6e20fd6e10c1b77654d067c0618f6e5a7f79a';
+				const invalidProof =
+					'b803eb0ed93ea10224a73b6b9c725796be9f5fefd215ef7a5b97234cc956cf6870db6127b7e4d824ec62276078e787db05584ce1adbf076bc0808ca0f15b73d59060254b25393d95dfc7abe3cda566842aaedf50bbb062aae1bbb6ef3b1fffff';
+				const context = createTransientModuleEndpointContext({
+					stateStore,
+					params: {
+						proofOfPossession: invalidProof,
+						blsKey: validPk,
+					},
+				});
+				await expect(validatorsModule.endpoint.validateBLSKey(context)).resolves.toStrictEqual({
+					valid: false,
+				});
+			});
+
+			it('should resolve with false when bls key length is less than 48 bytes and proof of possession has valid length', async () => {
+				const invalidPk = utils.getRandomBytes(47).toString('hex');
+				const validProof =
```

Comment:
Maybe we could makes the test cleaner by declaring repetitive vars globally to the suite and reuse them? As a result probably have less magic numbers in the tests. Otherwise everything else is ok 🙂